### DISCUSSION TITLE
Preferred citation format

### DISCUSSION THREAD

**Discussion by mylonasc** [OpenFAST/openfast#733] [General]
Currently there is no straightforward way to cite OpenFAST as a whole and give credit to the main authors of the code or the people that heavilly contributed with research on making it happen. 

This is more of a discussion than a feature request. 

Is there some default citation for openFAST as a whole or the modules should be individually adressed?

**Comment by jjonkman** [comment]
Good question.

In a recent paper, I referenced OpenFAST quite generically e.g.:

"OpenFAST (formerly known as FAST) [1] models the ...."

"REFERENCES
[1] Web page https://github.com/OpenFAST/openfast (accessed May 7, 2018)."

But I agree that this is very generic and doesn't really identify specific contributions.  Depending on what I want to refer to, I've also referenced specific OpenFAST-related publications e.g. here is the AIAA SciTech 2013 paper describing the FAST modularization framewok: 

"[4] Jonkman, J. M. “The New Modularization Framework for the FAST Wind Turbine CAE Tool.” 51st AIAA Aerospace Sciences Meeting including the New Horizons Forum and Aerospace Exposition, 7–10 January 2013, Grapevine (Dallas/Ft. Worth Region), TX [online proceedings]. URL: http://arc.aiaa.org/doi/pdf/10.2514/6.2013-202. AIAA-2013-0202. Reston, VA: American Institute of Aeronautics and Astronautics, January 2013; NREL/CP-5000-57228. Golden, CO: National Renewable Energy Laboratory."

But I'd be happy to hear what others recommend.

**Comment by rafmudaf** [comment]
What do you think about:
```
OpenFAST vX.Y.Z. Accessed May 7, 2018. Available at https://github.com/openfast/openfast/.
```
I think Google Scholar will track OpenFAST vX.Y.Z

**Comment by mylonasc** [comment]
I would suggest something more inclusive - for example, 4-5 authors/coders or people the namelist of people that contributed to approximately 90% of the whole code and functionality. I'm not very familiar with the contribution history of FAST and OpenFAST to provide such a name list.

**Comment by jjonkman** [comment]
I'm not sure it is common to reference a website / software in terms of lead developers.  I think the citations identified above are more common in literature.

Some of the biggest contributors to OpenFAST (and its predecessor, FAST) are (in alphabetical order, by last name) :
Marshall Buhl
Greg Hayman
Jason Jonkman
Bonnie Jonkman
Rafael Mudafort
Andy Platt
Mike Sprague

But there are many, many others who provided important contributions to the software over the years.

**Comment by ebranlard** [comment]
We could maybe add a entry on `zenodo` https://zenodo.org/ , that gives us a DOI to use for references. 

See for instance this guide: 
https://guides.github.com/activities/citable-code/#finishing-up

**Comment by rafmudaf** [comment]
So we've done this with [FLORIS](https://zenodo.org/record/3256660#.XbHewi2ZOXo) and I've been thinking about it for OpenFAST also, but I'm just not sure what it really gets us in addition to another reference to the software. I guess this would set the citation format, but I'm not really a fan of its method. For FLORIS, zenodo says:

```
Rafael M Mudafort, paulf81, bayc, Pete Bachant, ejsimley, Katherine Fleming, & jrannoni. (2019, June 25). NREL/floris: v1.1.4 (Version v1.1.4). Zenodo. http://doi.org/10.5281/zenodo.3256660
```

In this case, `paulf81` and `bayc` are GitHub usernames and the authors are listed in order of most contributions.

**Comment by ebranlard** [comment]
I see that it's indeed not the best output... I'd think that as long as you provide the bibtex entry for the latex users and the preferred citation string for other users, it would be fine. For the bibtex entry, you can add  `doi = {XX.XXXX/zenodo.XXXXXXX}`.  But that's just a suggestion. 

Last time I used the suggested citation format with the github repository, the journal I submitted to replaced it by: ` Available at http://doi.org/github.com/OpenFAST/OpenFAST/`  which is not ideal either (maybe I should have paid attention to it.. )

**Reply by jjonkman** [reply]
For publication in WES, I was told I needed a DOI for all software, so, I used zenodo for OpenFAST, which came up with: 

Jonkman, B., Mudafort, R. M., Platt, A., Branlard, E., Sprague, M., Jonkman, J., Hayman, G., Vijayakumar, G., Buhl, M., Ross, H.,
Bortolotti, P., Masciola, M., Ananthan, S., Schmidt, M. J., Rood, J., Mendoza, N., Hai, S. L., Hall, M., Sharma, A., Shaler, K., Bendl, K., Schuenemann, P., Sakievich, P., Quon, E.W., Phillips, M. R., Kusouno, N., Gonzalez Salcedo, A., Martinez, T., andCorniglion, R.: OpenFAST/openfast: OpenFAST v3.1.0, Zenodo [code], https://doi.org/10.5281/zenodo.6324288, 2022.

**Comment by Cardoza2** [comment]
I don't know if it's within the scope of your work anymore, but have you guys considered submitting something to the [Journal of Open Source Software](https://joss.theoj.org/about) ? I don't know tons about it, but you'd have a citation the code base.

**Reply by rafmudaf** [reply]
I've considered it but haven't gotten around to going through the submission process. I'll be doing this for https://github.com/nrel/floris in August/September, so that may be an opportunity to do it for OpenFAST, as well.
cc @andrew-platt @ebranlard @deslaughter
